File: gestionPatient/src/main/java/com/gestionpatient/entities/RendezVous.java
package com.gestionpatient.entities;



import jakarta.persistence.*;
import jakarta.persistence.Id;
import lombok.*;
import org.springframework.data.annotation.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="RendezVous")
@EntityListeners(AuditingEntityListener.class)
public class RendezVous  {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long idRdv;

    @Column(name = "codeRdv")
    private String codeRdv;

    @Column(name = "dateRdv")
    private Date dateRdv;

    @Column(name = "heureRdv")
    private String heureRdv;

    @Column(name = "idPatient")
    private Long idPatient;

    @Column(name = "statusRdv" , length = 3)
    private String statusRdv;

    @Column(name="typeRdv")
    private String typeRdv;

    @Column(name="motif")
    private String motif;

    @CreatedDate
    @Column(name = "createdAt")
    private Date createdAt;

    @LastModifiedDate
    @Column(name = "modifiedAt")
    private Date modifiedAt;

    @ManyToOne
    @JoinColumn(name="idPatient", referencedColumnName = "idPatient", nullable=false, insertable = false, updatable = false)
    private Patient patient;



}

File: MyClinic/src/app/features/models/rendez-vous.ts
import { Patient } from "./patient";

export class RendezVous {
  public static fromJson(json:any):RendezVous{
    return new RendezVous(
      json['idRdv'],
      json['codeRdv'],
      json['dateRdv'],
      json['heureRdv'],
      json['idPatient'],
      json['statusRdv'],
      json['typeRdv'],
      json['motif'],
      json['createdAt'],
      json['modifiedAt'],
      json['patient'],
    )
  }constructor(public idRdv: number,
                public codeRdv: string,
                public dateRdv: Date,
                public heureRdv: string,
                public idPatient: number,
                public statusRdv: string,
                public typeRdv:string,
                public motif:string,
                public createdAt: Date,
                public modifiedAt: Date,
                public patient:Patient){}
}
